kornia/geometry/camera/projection_z1.py

```diff
@@ -0,0 +1,104 @@
+"""Module for the projection of points in the canonical z=1 plane."""
+# inspired by: https://github.com/farm-ng/sophus-rs/blob/main/src/sensor/perspective_camera.rs
+from __future__ import annotations
+
+from typing import Optional
+
+import kornia.core as ops
+from kornia.core import Tensor
+from kornia.core.check import KORNIA_CHECK_SHAPE
+
+
+def project_points_z1(points_in_camera: Tensor) -> Tensor:
+    r"""Project one or more points from the camera frame into the canonical z=1 plane through perspective division.
+
+    .. math::
+
+        \begin{bmatrix} u \\ v \\ w \end{bmatrix} =
+        \begin{bmatrix} x \\ y \\ z \end{bmatrix} / z
+
+    Args:
+        points_in_camera: Tensor representing the points to project with shape (..., 3).
+
+    Returns:
+        Tensor representing the projected points with shape (..., 2).
+
+    Example:
+        >>> points = torch.tensor([1., 2., 3.])
+        >>> project_points_z1(points)
+        tensor([0.3333, 0.6667])
+    """
+    KORNIA_CHECK_SHAPE(points_in_camera, ["*", "3"])
+    return points_in_camera[..., :2] / (points_in_camera[..., 2:3] + 1e-08)
+
+
+def unproject_points_z1(points_in_cam_canonical: Tensor, extension: Optional[Tensor] = None) -> Tensor:
+    r"""Unproject one or more points from the canonical z=1 plane into the camera frame.
+
+    .. math::
+        \begin{bmatrix} x \\ y \\ z \end{bmatrix} =
+        \begin{bmatrix} u \\ v \end{bmatrix} \cdot w
+
+    Args:
+        points_in_cam_canonical: Tensor representing the points to unproject with shape (..., 2).
+        extension: Tensor representing the extension of the points to unproject with shape (..., 1).
```

Comment:
perhaps we should specify `depth` in brackets(`extension(depth):`) for more clarity?

Reply:
done